src/hotspot/share/services/memReporter.cpp

```diff
@@ -50,10 +50,13 @@ size_t MemReporterBase::committed_total(const MallocMemory* malloc, const Virtua
   return malloc->malloc_size() + malloc->arena_size() + vm->committed();
 }
 
-void MemReporterBase::print_total(size_t reserved, size_t committed) const {
+void MemReporterBase::print_total(size_t reserved, size_t committed, size_t peak) const {
   const char* scale = current_scale();
   output()->print("reserved=" SIZE_FORMAT "%s, committed=" SIZE_FORMAT "%s",
     amount_in_current_scale(reserved), scale, amount_in_current_scale(committed), scale);
+  if (peak != 0) {
+    output()->print(", largest_committed=" SIZE_FORMAT "%s", amount_in_current_scale(peak), scale);
+  }
```

Comment:
This seems odd. Not a problem of the backport, but of the original change. For any value lower than `scale` we'll get `0` as a result if I'm reading this right.

Your example output showed:
```
(mmap: reserved=8388608KB, committed=507904KB, largest_committed=0KB) 
```

Yet, it shouldn't be there (due to `peak != 0`). We should get this fixed before we backport.

Reply:
Ok, I'll investigate

Reply:
For the record this will be addressed by also backporting [JDK-8320061](https://bugs.openjdk.org/browse/JDK-8320061). PR #481 depending on this one.

Reply:
yes, in Marias example output, largest_committed should be >= committed. That issue has been indeed fixed with [JDK-8320061](https://bugs.openjdk.org/browse/JDK-8320061).

The "amount_in_current_scale results in zero for very small numbers" can still happen here and there, but its not the case in this example.